refactor: replace lodash & assertive usage

aaarichter opened the pull request:
- [x] refactor: remove lodash usage
- [x] refactor: use node assert
- [x] test: rewrite test with native assert 
- [x] chore: update packages

dbushong commented on lib/browser/_asserters.js:
```diff
@@ -105,19 +105,22 @@ exports.fuzzyString = function assertFuzzyString(
 
   return new Asserter(target => {
     return Promise.resolve(target)
-      .then(extract)
+      .then(res => {
+        return extract(res);
+      })
```
out of curiosity - why this change?

aaarichter replied:
💅  was testing something there and didn't revert

dbushong commented on lib/browser/cookie.js:
```diff
@@ -66,19 +63,24 @@ exports.setCookies = util.deprecate(function setCookies(cookies) {
   return Promise.all(
-    _.map(cookies, (value, name) => {
-      return browser.setCookieValue(name, value);
+    Object.entries(cookies).map(pair => {
+      return browser.setCookieValue(...pair);
```
minor nit: while this is clever, if you ever end up wanting to type check this code, it'll work better to have something more explicit like:
```suggestion
    Object.entries(cookies).map(([k, v]) => {
      return browser.setCookieValue(k, v);
```

dbushong commented on lib/browser/lighthouse.js:
```diff
@@ -33,8 +33,9 @@
-const flatMap = require('lodash/flatMap');
+const flatMap = require('array.prototype.flatmap');
+
+flatMap.shim(); // TODO: remove with node 12
```
I would recommend using it directly as a ponyfill - it's safer and not that much harder to refactor later.

dbushong commented on lib/testium-driver-wd.js:
```diff
@@ -194,19 +201,19 @@ function initDriver(testium) {
+        throw new Error('Lighthouse requires a newer version of node');
+      }
+      const devtoolsPort = testium.getChromeDevtoolsPort();
+      const options = Object.assign({ port: devtoolsPort }, flags);
```
```suggestion
      const options = { port: devtoolsPort , ...flags };
```

dbushong commented on test/integration/dialog.test.js:
```diff
@@ -27,60 +27,102 @@ describe('dialogs', () => {
-      const text = await browser.getAlertText();
-      await browser.acceptAlert();
-      assert.equal('Alert text was not found', 'An alert!', text);
+    it('can get an alert text', done => {
```
Why'd these all change back to testing callback mode?

aaarichter replied:
I discovered that `acceptAlert()` and `dismissAlert()` are not part of the promise chain, but use WD commands directly.
Locally those tests run fine with `aync/await` but they are somewhat flaky in Travis

I'll promisify the them in a follow up.

dbushong reviewed (approved):
Except for the polyfill and the callback test conversion commented on, LGTM!

aaarichter commented:
Thanks for the review @dbushong . I addressed your feedback, but kept the prompt test changes for now. As mentioned, in the follow up, I'll promisify them, and revert the tests to the previous async await tests
